### PR TITLE
Allow switching data sources

### PR DESCRIPTION
**Pull request by kjordahl**
Introduce multiple data sources. Currently can toggle between forecast and hindcast.

Not quite complete, needs to update timestamps on switching data sources, and clean up a few more things.

### CODE REVIEW

File: tabs/thredds_frame_source.py [hunk 1]
```diff
@@ -189,7 +191,7 @@ def main(NFRAMES=90, output_dir=None):
         output_dir = os.path.join(os.path.dirname(__file__),
                                   '../static/data/json')
     filename = partial(os.path.join, output_dir)
-    frame_source = THREDDSFrameSource(DEFAULT_DATA_URI, decimate_factor=10)
+    frame_source = THREDDSFrameSource(HINDCAST_CACHE_DATA_URI, decimate_factor=10)
```
Comment (johntyree) [inline tabs/thredds_frame_source.py]: Didn't we want to default to the live connection?

**Comment by johntyree** [comment]
Can you also update the script that pulls down the netcdf files for the cached data to get this new one?

File: tabs/thredds_frame_source.py (hunk 1 above)
Comment (kjordahl) [inline tabs/thredds_frame_source.py]: I's always passed explicitly in the thredds connection, so this only applies to the standalone test of the connection. I don't think it matters much here.

**Comment by kjordahl** [comment]
> Can you also update the script that pulls down the netcdf files for the cached data to get this new one?

Will do. Also the `--cached` flag is broken, because it doesn't actually use them.

**Comment by johntyree** [comment]
Yes, I was just looking at fixing that now.

**Comment by kjordahl** [comment]
One bug that needs to be fixed before merging this is that going past the end of the forecast data kills it. Should handle 404's gracefully.

**Comment by kjordahl** [comment]
Actually working now, with minor glitches remaining. The number of time steps switches correctly, but the length of the slider doesn't reflect that, for example.

**Comment by kjordahl** [comment]
It still doesn't handle 404s gracefully, it just avoids them in the common case of switching to a data source with fewer steps available.

**Comment by kjordahl** [comment]
Set the default number of frames to 30 for now for consistency with the forecast data. This makes switching back and forth easier. Changing the number of frames will need some work to update the slider (otherwise it would still be possible to drag it past the end of the data).

I think this is ready to merge now.

**Comment by johntyree** [comment]
LGTM

**Comment by johntyree** [comment]
Except for the bug i just introduced.

**Pull request merged**